metricbeat/module/kubernetes/util/metrics_storage.go

```diff
@@ -0,0 +1,200 @@
+package util
+
+import (
+	"strings"
+	"sync"
+)
+
+type Metrics struct {
+	entries map[Metric]float64
+}
+
+func NewMetrics() *Metrics {
+	ans := &Metrics{
+		entries: make(map[Metric]float64),
+	}
+	return ans
+}
+
+func (m *Metrics) Set(name Metric, value float64) {
+	m.entries[name] = value
+}
+
+func (m *Metrics) Get(name Metric) (float64, bool) {
+	ans, exists := m.entries[name]
+	return ans, exists
+}
+
+func (m *Metrics) GetWithDefault(name Metric, defaultValue float64) (float64, bool) {
+	ans, exists := m.Get(name)
+	if !exists {
+		return defaultValue, false
+	}
+	return ans, exists
+}
+
+func (m *Metrics) Delete(name Metric) {
+	delete(m.entries, name)
+}
+
+func (m *Metrics) Clear() {
+	for k := range m.entries {
+		delete(m.entries, k)
+	}
+}
+
+type MetricSet struct {
+	sync.RWMutex
+	metrics map[string]*Metrics
+}
+
+func NewMetricSet() *MetricSet {
+	ans := &MetricSet{
+		metrics: make(map[string]*Metrics),
+	}
+	return ans
+}
+
+func (ms *MetricSet) Clear() {
+	ms.Lock()
+	defer ms.Unlock()
+	for k := range ms.metrics {
+		delete(ms.metrics, k)
+	}
+}
+
+func (ms *MetricSet) addMetrics(id string) *Metrics {
+	ms.Lock()
+	defer ms.Unlock()
+	ms.metrics[id] = NewMetrics()
+	return ms.metrics[id]
+}
+
+func (ms *MetricSet) getMetrics(id string) (*Metrics, bool) {
+	ms.RLock()
+	defer ms.RUnlock()
+	ans, exists := ms.metrics[id]
+	return ans, exists
+}
+
+func (ms *MetricSet) Get(id string, metricName Metric) (float64, bool) {
```

Comment:
There is at least one more `Get`  method in this file. This makes it hard to understand which is one should be exposed or not and what it the hierarchy. It would be nice to have them organised in way that makes reading easy and straightforward.

Reply:
after my latest changes we now have a Metric.Get and a MetricsStorage.Get. The difference between the two is:
- Metric.Get returns a metric by name
- MetricsStorage.Get return a set of metrics by metric id

I'll add a doc string and some extra documentation in the code but I am not sure how to make it `reading easy and straightforward`. Any suggestions